### PR TITLE
Two (more) environment fixes

### PR DESCRIPTION
**Pull request by dra27**
Two fixes here, both relatively subtle:

- If the environment contained `FOO=bar`, the update `FOO += ""` would cause opam to execute _without_ `FOO` in the environment
- A small regression in #5356 - in the refactoring for case-preserving updates, a boolean check got accidentally inverted

### CODE REVIEW

**Review by rjbou — COMMENTED** [review]
It is missing a test for the case *add an empty string to an undefined variable*. Nothing changes in the environment as the variable is not defined beforehand.
```diff
diff --git a/tests/reftests/env.test b/tests/reftests/env.test
index 0a8a7a210..781341de3 100644
--- a/tests/reftests/env.test
+++ b/tests/reftests/env.test
@@ -80,6 +80,7 @@ setenv: [
   [ NV_VARS3 := ""    ]
   [ NV_VARS3 := "foo" ]
   [ NV_VARS4  = ""    ]
+  [ NV_VARS5 += ""    ] # undefined
 ]
 flags: compiler
 ### opam switch create emptyvar nv
```
An update of the documentation to be more precise, and lgtm!

**Comment by dra27** [comment]
I think the manual’s correct, isn’t it? The issue was that opam wasn’t previously doing what it was supposed to (by any definition of `+=`)

**Review by rjbou — APPROVED** [review]
thanks!

**Comment by kit-ty-kate** [comment]
Thanks
